tools/esdlcmd/esdlcmd_monitor.cpp

```diff
@@ -1001,39 +914,44 @@ class EsdlDiffGenCmd : public EsdlConvertCmd
 
         respTemplate = monitoringTemplate->queryPropTree(resp_type);
 
-        xpath.setf("EsdlStruct[@name='%s']", resp_type.str());
-        IPropertyTree *respTree = depTree->queryPropTree(xpath);
-        if (respTree)
-        {
-            respTree = checkExtractNestedResponseType(depTree, respTree, resp_type);
+        IPropertyTree *respTree = depTree->queryPropTree(xpath.setf("EsdlResponse[@name='%s']", resp_type.str()));
```

Comment:
I'm surprised PropertyTree::queryPropTree accepts a StringBuffer, there's no need to call .str() on xpath?